docker-compose.test.yml

```diff
@@ -0,0 +1,25 @@
+
```

Comment:
In the local setup, it is important to run the BIT backend first before running the MS backend. This is to make sure that the postgresql database schemas are build correctly using the db models from BIT backend code. 
In this scenarion (running the app on docker), how do you make sure that this constraint be followed? Should we add build BIT 1st, wait, then build MS next?

Reply:
Also, how to inspect postgres db? can we still use the pgadmin4 to connect to this? If we can, please add this in the setup instruction for docker.

Reply:
Also, if we want to debug, can you please add instruction on docker on how to do this (command to be run for docker to run in debug mode)?

Reply:
> I tried to run docker by running `docker-compose up` and tried to register, but failed. Possible cause is MS being built and run first before BIT which messed up the schemas inside postgres db.
> [Here's](https://gist.github.com/mtreacy002/eccb93ef0dc0c4677ab84c424806e926) the terminal log.

I think it is because you set the postgres host to localhost. It should be set to `postgers` while running the app and to `test_postgres` while testing it ( If you are using docker).

Reply:
> In the local setup, it is important to run the BIT backend first before running the MS backend. This is to make sure that the postgresql database schemas is built correctly using the db models from BIT backend code.
> In this scenario (running the app on docker), how do you make sure that this constraint being followed? Should we add build BIT 1st, wait, then build MS next?

The schemas are actually initialized when the postgres container is built using the script `init_database.sql`. But if you are talking about tables which are built are runtime there is not way to ensure that the tables from bit are created first : / .

Reply:
> Also, how to make sure the docker runs the correct branch on MS and BIT on:
> 
> * normal situation: BIT == develop branch, MS == bit branch (assuming the PR is approved and merged)
> * working on PR: BIT == pr branch, MS == bit branch

We can control the branch of the MS backend once it is merged however for the BIT backend it would just be the branch that the contributor has checked out.

Reply:
> Also, how to inspect postgres db? can we still use the pgadmin4 to connect to this? or we must use command line to inspect potgres inside docker? whichever options are possible, can you please explain this in the setup instruction for docker.

You can connect to postgers using the docker command line. There are add-ons that let you use pgadmin4 for the database inside docker. Would you like me to set that up?

Reply:
> Also, if we want to debug, can you please add instruction on docker on how to do this (command to be run for docker to run in debug mode)?

Sure.

Reply:
> I think it is because you set the postgres host to localhost. It should be set to `postgers` while running the app and to `test_postgres` while testing it ( If you are using docker).

So I need to add these extra variables inside my local .env? I thought that isn't necessary as docker has it's own env when it sets up postgres within docker? I'm confused 🤔. I looked up POSTGRES_HOST and only found it being setup inside Dockerfile.
<img width="387" alt="Screen Shot 2021-03-10 at 6 15 30 pm" src="https://user-images.githubusercontent.com/29667122/110591783-97345480-81cd-11eb-97a8-eac1fca6b8a0.png">

>  It should be set to `postgers` while running the app and to `test_postgres` while testing it ( If you are using docker)
Can you please be specific on how and where exactly we need to do this (set to postgres and test_postgres while running the app using docker)

Reply:
> > Also, how to inspect postgres db? can we still use the pgadmin4 to connect to this? or we must use command line to inspect potgres inside docker? whichever options are possible, can you please explain this in the setup instruction for docker.
> 
> You can connect to postgers using the docker command line. There are add-ons that let you use pgadmin4 for the database inside docker. Would you like me to set that up?

Yes please

Reply:
> The bit backend is working fine but the issue is with the MS backend (as can be seen in the error log). I have sent a pr to your `initial-bit-integration` branch [here](https://github.com/mtreacy002/mentorship-backend/pull/23) you'll have to use this branch. Also remember to clean up the containers now that the code has changed remember to rebuild the containers. (You can use `make docker-dev` and it will handle all of this for you : ) )

you mean the command `make docker_host_dev` as in the Dockerfile, right?